src/test/app/Check_test.cpp

```diff
@@ -1848,22 +1883,654 @@ class Check_test : public beast::unit_test::suite
+        }
+        //------------ lsfGlobalFreeze, check written by non-issuer ------------
+        {
+            // Since gw1 has the lsfGlobalFreeze flag set, there should be
+            // no automatic trust line creation between non-issuers.
+
+            // Use offers to automatically create the trust line.
+            IOU const OF4 = gw1["OF4"];
+            env(offer(alice, XRP(91), OF4(91)), ter(tecFROZEN));
+            env.close();
+            BEAST_EXPECT(
+                env.le(keylet::line(alice, bob, OF4.currency)) == nullptr);
+            env(offer(bob, OF4(91), XRP(91)), ter(tecFROZEN));
+            env.close();
+
+            // No one's owner count should have changed.
+            BEAST_EXPECT(ownerCount(env, gw1) == 0);
+            BEAST_EXPECT(ownerCount(env, alice) == 6);
+            BEAST_EXPECT(ownerCount(env, bob) == 5);
+
+            // Use check cashing to automatically create the trust line.
+            IOU const CK4 = gw1["CK4"];
+            uint256 const chkId{getCheckIndex(alice, env.seq(alice))};
+            env(check::create(alice, bob, CK4(91)), ter(tecFROZEN));
+            env.close();
+            BEAST_EXPECT(
+                env.le(keylet::line(alice, bob, CK4.currency)) == nullptr);
+            env(check::cash(bob, chkId, CK4(91)), ter(tecNO_ENTRY));
+            env.close();
+
+            // No one's owner count should have changed.
+            BEAST_EXPECT(ownerCount(env, gw1) == 0);
+            BEAST_EXPECT(ownerCount(env, alice) == 6);
+            BEAST_EXPECT(ownerCount(env, bob) == 5);
+
+            // Because gw1 has set lsfGlobalFreeze, neither trust line
+            // is created.
+            BEAST_EXPECT(
+                env.le(keylet::line(gw1, bob, OF4.currency)) == nullptr);
+            BEAST_EXPECT(
+                env.le(keylet::line(gw1, bob, CK4.currency)) == nullptr);
+        }
+
+        //-------------- lsfRequireAuth, check written by issuer ---------------
+
+        // We want to test the lsfRequireAuth flag, but we can't set that
+        // flag on an account that already has trust lines.  So we'll fund
+        // a new gateway and use that.
+        Account const gw2{"gw2"};
+        env.fund(XRP(5000), gw2);
+        env.close();
+        {
+            // Set lsfRequireAuth on gw2.  That should stop any automatic
+            // trust lines from being created.
+            env(fset(gw2, asfRequireAuth));
+            env.close();
+
+            // Use offers to automatically create the trust line.
+            IOU const OF5 = gw2["OF5"];
+            env(offer(gw2, XRP(92), OF5(92)));
+            env.close();
+            BEAST_EXPECT(
+                env.le(keylet::line(gw2, alice, OF5.currency)) == nullptr);
+            env(offer(alice, OF5(92), XRP(92)), ter(tecNO_LINE));
+            env.close();
+
+            // No one's owner count should have changed.
+            BEAST_EXPECT(ownerCount(env, gw1) == 0);
```

Comment:
This should be `gw2`. It might be worth moving `gw1` into a new separate scope for the tests above to ensure this code can't see it.